src/codeu/chat/client/commandline/Chat.java

```diff
@@ -173,6 +175,20 @@ public void invoke(List<String> args) {
       }
     });
 
+    panel.register("ui-add", new Panel.Command() {
+      @Override
+      public void invoke(List<String> args) {
+        final String name = args.size() > 0 ? args.get(0) : "";
+        if (name.length() > 0) {
+          if (context.add(name) == null) {
+            System.out.println("ERROR: Failed to create new user interest");
+          }
+        } else {
+          System.out.println("ERROR: Missing <username>");
+        }
+      }
+    });
```

Comment:
You need to move ui-add to the  user context panel below, same place where ci-add is specified, because you actually need to call beingNewUserInterest on the UserContext. There should be no new methods on Context.